### ISSUE TITLE
Misc. bug: Disable KV cache offload command line parameter not being honored. -nkvo Recent regression.

### ISSUE THREAD

**Issue by bitbottrap** [ggerganov/llama.cpp#10494]
### Name and Version

ggml_cuda_init: GGML_CUDA_FORCE_MMQ:    no
ggml_cuda_init: GGML_CUDA_FORCE_CUBLAS: no
ggml_cuda_init: found 2 CUDA devices:
  Device 0: NVIDIA GeForce RTX 3090 Ti, compute capability 8.6, VMM: yes
  Device 1: NVIDIA GeForce RTX 3090 Ti, compute capability 8.6, VMM: yes
version: 4164 (9ca2e677)
built with cc (Ubuntu 13.2.0-23ubuntu4) 13.2.0 for x86_64-linux-gnu

### Which operating systems do you know to be affected?

Linux

### Which llama.cpp modules do you know to be affected?

llama-server

### Steps to Reproduce

-nkvo doesn't appear to be processed correctly anymore. Seeing offloading of the kv cache even when explicitly disabled using the following command line:
llama-server --host 0.0.0.0 --port 8091 --threads 60 --numa isolate -nkvo -ngl 0 -c 131072 --flash-attn --cache-type-k q8_0 --cache-type-v q8_0 -m /models/llama3.1:405b-instruct-q8_0.gguf

I believe this was working roughly a week or two ago without issue.

### Relevant log output

```shell
Nov 25 10:03:07 sv-ai llama-server[174008]: llama_new_context_with_model: n_seq_max     = 1
Nov 25 10:03:07 sv-ai llama-server[174008]: llama_new_context_with_model: n_ctx         = 131072
Nov 25 10:03:07 sv-ai llama-server[174008]: llama_new_context_with_model: n_ctx_per_seq = 131072
Nov 25 10:03:07 sv-ai llama-server[174008]: llama_new_context_with_model: n_batch       = 2048
Nov 25 10:03:07 sv-ai llama-server[174008]: llama_new_context_with_model: n_ubatch      = 512
Nov 25 10:03:07 sv-ai llama-server[174008]: llama_new_context_with_model: flash_attn    = 1
Nov 25 10:03:07 sv-ai llama-server[174008]: llama_new_context_with_model: freq_base     = 500000.0
Nov 25 10:03:07 sv-ai llama-server[174008]: llama_new_context_with_model: freq_scale    = 1
Nov 25 10:03:26 sv-ai llama-server[174008]: llama_kv_cache_init:        CPU KV buffer size = 34272.00 MiB
Nov 25 10:03:26 sv-ai llama-server[174008]: llama_new_context_with_model: KV self size  = 34272.00 MiB, K (q8_0): 17136.00 MiB, V (q8_0): 17136.00 MiB
Nov 25 10:03:26 sv-ai llama-server[174008]: llama_new_context_with_model:        CPU  output buffer size =     0.49 MiB
Nov 25 10:03:27 sv-ai llama-server[174008]: llama_new_context_with_model:      CUDA0 compute buffer size =  2411.75 MiB
Nov 25 10:03:27 sv-ai llama-server[174008]: llama_new_context_with_model:  CUDA_Host compute buffer size =   288.01 MiB
Nov 25 10:03:27 sv-ai llama-server[174008]: llama_new_context_with_model: graph nodes  = 3535
Nov 25 10:03:27 sv-ai llama-server[174008]: llama_new_context_with_model: graph splits = 1391 (with bs=512), 1 (with bs=1)
```

**Comment by slaren** [comment]
You can see here that the KV is allocated in a CPU buffer:

> ```shell
> Nov 25 10:03:26 sv-ai llama-server[174008]: llama_kv_cache_init:        CPU KV buffer size = 34272.00 MiB
> ```

**Comment by bitbottrap** [comment]
You can see the compute buffer allocated on a GPU buffer here:
Nov 25 10:03:27 sv-ai llama-server[174008]: llama_new_context_with_model:      CUDA0 compute buffer size =  2411.75 MiB
Nov 25 10:03:27 sv-ai llama-server[174008]: llama_new_context_with_model:  CUDA_Host compute buffer size =   288.01 MiB

Might have misidentified which flag was responsible then. Which flag controls the compute buffers, KV cache or ngl?

There should be nothing allocated on the GPU with both of these flags in use.

**Comment by slaren** [comment]
Computation of large batches is still offloaded. If you want to disable GPU usage entire, you can do that by setting the environment variable `CUDA_VISIBLE_DEVICES` to an empty string.

**Comment by bitbottrap** [comment]
Okay, will do. This is a change in behavior. Up until recently nothing was allocated on the GPU. Thank you for your response.

**Comment by slaren** [comment]
It has worked this way pretty much since the first day support for CUDA was added.

**Comment by bitbottrap** [comment]
It's difficult for me to argue the point but nvidia-smi had not reported any GPU VRAM usage for this process, with the exact same command line, until recently. Been running it for over a month. I noticed this when I started to running out of GPU VRAM to load the configuration I usually load (several llama server processes serving different models) and noticed the new GPU VRAM usage from these processes. In any case, I'll use the proper method to suppress this. I thank you for replying!
